torchvision/prototype/transforms/_misc.py

```diff
@@ -140,14 +140,25 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
         return F.gaussian_blur(inpt, self.kernel_size, **params)
 
 
-# TODO: Enhance as described at https://github.com/pytorch/vision/issues/6697
-class ToDtype(Lambda):
-    def __init__(self, dtype: torch.dtype, *types: Type) -> None:
+class ToDtype(Transform):
+    def __init__(self, dtype: Union[torch.dtype, OrderedDictType[Type, torch.dtype]]) -> None:
+        super().__init__()
+        if isinstance(dtype, torch.dtype):
+            dtype = OrderedDict([(torch.Tensor, dtype)])
         self.dtype = dtype
-        super().__init__(functools.partial(torch.Tensor.to, dtype=dtype), *types or (torch.Tensor,))
 
-    def extra_repr(self) -> str:
-        return ", ".join([f"dtype={self.dtype}", f"types={[type.__name__ for type in self.types]}"])
+    def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
+        input_type = type(inpt)
+        if input_type in self.dtype:
+            dtype = self.dtype[input_type]
+        else:
+            for to_type, dtype in self.dtype.items():
+                if issubclass(input_type, to_type):
+                    break
+            else:
+                return inpt
```

Comment:
This firsts tests if we have an exact type match. If we don't, we go through the dictionary and see if we find a subclass match and use that. This is useful in case someone wants to just convert all tensors to a specific `dtype`. Since we can't know what exact types we might encounter at runtime, we use `torch.Tensor` as default type that will match all tensors at the second stage above even if the tensors are features.

Reply:
Why not go for direct types as we have on `fill`? I'm a bit skeptical for adding this extra logic here as it diverges from what we do elsewhere. I think we should align by just checking for `type(inpt)` and leaving it to the user to define things properly. WDYT?